File: examples/py-calc/calc.py
"""A tiny calculator, used to demo Master-Anything's verifiable-mastery loop."""


class Calculator:
    def add(self, a, b):
        return a + b

    def sub(self, a, b):
        return a - b

    def add_many(self, nums):
        total = 0
        for n in nums:
            total = self.add(total, n)
        return total


def average(nums):
    calc = Calculator()
    if not nums:
        return 0
    return calc.add_many(nums) / len(nums)

File: examples/js-calc/calc.js
// A tiny calculator in JS, for the JavaScript verifiable-mastery demo.

class Calculator {
  add(a, b) {
    return a + b;
  }

  sub(a, b) {
    return a - b;
  }

  addMany(nums) {
    let total = 0;
    for (const n of nums) {
      total = this.add(total, n);
    }
    return total;
  }
}

function average(nums) {
  if (nums.length === 0) return 0;
  return new Calculator().addMany(nums) / nums.length;
}

module.exports = { Calculator, average };
